pandas/_libs/reduction.pyx

```diff
@@ -603,7 +603,7 @@ cdef class BlockSlider:
             arr.shape[1] = 0
 
 
-def compute_reduction(arr: np.ndarray, f, axis: int = 0, dummy=None, labels=None):
+def compute_reduction(arr: ndarray, f, axis: int = 0, dummy=None, labels=None):
```

Comment:
mypy is not checking .pyx files. I suspect we would want to keep the np.ndarray annotation here since ndarray is a cimport.